rpcs3/rpcs3qt/gui_application.cpp

```diff
@@ -569,6 +569,53 @@ void gui_application::InitializeCallbacks()
 		};
 	}
 
+	callbacks.on_emulation_stop_no_response = [this](std::shared_ptr<atomic_t<bool>> closed_successfully)
+	{
+		Emu.CallFromMainThread([this, closed_successfully]
+		{
+			const auto seconds = std::make_shared<int>(10);
+
+			QMessageBox* mb = new QMessageBox();
+			mb->setWindowTitle(tr("PS3 Game/Application Is Unresponsive"));
+			mb->setIcon(QMessageBox::Critical);
+			mb->setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+			mb->setDefaultButton(QMessageBox::No);
+			mb->button(QMessageBox::Yes)->setText(tr("Close Unsafely"));
+			mb->button(QMessageBox::No)->setText(tr("Keep Waiting"));
+
+			QString text_base = tr("Waiting for %0 seconds already to stop emulation without success."
+									"\nDo you want to wait more for it to safely stop or to stop it unsafely now?");
+
+			mb->setText(text_base.arg(10));
+			mb->layout()->setSizeConstraint(QLayout::SetFixedSize);
+			mb->setAttribute(Qt::WA_DeleteOnClose);
+
+			QTimer* update_time = new QTimer(mb);
+			connect(update_time, &QTimer::timeout, [mb, seconds, text_base, closed_successfully]()
+			{
+				*seconds += 1;
+				mb->setText(text_base.arg(*seconds));
+
+				if (*closed_successfully)
+				{
+					mb->reject();
+				}
+			});
+
+			update_time->start(1000);
+
+			connect(mb, &QDialog::accepted, mb, [closed_successfully]
+			{
+				if (!*closed_successfully)
+				{
+					std::abort();
```

Comment:
this isn't conveyed in the message. The message sounds like you're just stopping the game.

Reply:
Called "Force Abort" now.